packages/agreement-readmodel-writer/test/agreement-readmodel-writer.V2.integration.test.ts

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import {
+  AgreementCollection,
+  ReadModelRepository,
+  readModelWriterConfig,
+} from "pagopa-interop-commons";
+import {
+  getMockAgreement,
+  mongoDBContainer,
+} from "pagopa-interop-commons-test";
+import {
+  AgreementEventEnvelopeV2,
+  generateId,
+  toAgreementV2,
+  toReadModelAgreement,
+} from "pagopa-interop-models";
+import { StartedTestContainer } from "testcontainers";
+import {
+  afterAll,
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  vi,
+} from "vitest";
+import { handleMessageV2 } from "../src/consumerServiceV2.js";
+
+describe("events V2", async () => {
+  let agreements: AgreementCollection;
+  let startedMongoDBContainer: StartedTestContainer;
+
+  const config = readModelWriterConfig();
+  beforeAll(async () => {
+    startedMongoDBContainer = await mongoDBContainer(config).start();
+
+    config.readModelDbPort = startedMongoDBContainer.getMappedPort(27017);
+
+    const readModelRepository = ReadModelRepository.init(config);
+    agreements = readModelRepository.agreements;
```

Comment:
After #441 you could just call 

```
const { readModelRepository } =
  setupTestContainersVitest(inject("config"));

agreements = readModelRepository.agreements;
```

It should take care of everything: initializing the mongo container, the readmodel, etc.

Reply:
The same function also returns a `cleanup` method in the same returned object. 
You must then use `cleanup` in `afterEach`

`afterEach(cleanup)`

No need for `afterAll` anymore, the container part is taken care of in the test setup already

Reply:
The tests here are not compatible with the new test setup because here we only want to start up the mongodb test container, but the new setup requires at least both mondob and postgres test container config in order to start.

The solution is to make the postgres config optional and start up the related test container only if they are present in the env. What do you think?

Reply:
Yep, didn't realize it, I did it in #509 and opened https://pagopa.atlassian.net/browse/IMN-536 to track the fact that we want to port the new setup to all services. @sandrotaje ignore the global test reuse comments for this PR :)